### DISCUSSION TITLE
Flickr tags to Labels and Keywords

### DISCUSSION THREAD

**Discussion by bcdixit** [photoprism/photoprism#1742] [General]
Hello guys,

Recently, I discovered this excellent tool after experimenting with Piwigo. 

had a question on photoprism keywords/labels.

I want to move all my flickr photos to Photoprism, but want to maintain all the tags that I have on flickr photos.
I downloaded all off my 12000+ photos (and videos) from flickr along with the the associated sidecar json files.

Then, for few sample photos, i embedded the tags from flickr into the IPTC:keywords tag using exiftool directly into the photos

When i moved few photos into Photoprism, I noticed that the "keywords" field does get populated but not all keywords from the photo are shown there.
Initially, i thought there was a 64char limit on the IPTC keywords, but i mitigated that by using the '-m' option in exiftool.

When I check the exifdata using exiftool and irfanview , I can see all the keywords that were added  directly to the photo.

**Is there a limit to the number of keywords per photo in Photoprism?**

Should I be using the Labels instead? 
Is there a programatic way to create multiple lables per photo? using API / Python scripts / SQL inserts/updates?

**Comment by graciousgrey** [comment]
Yes, currently there is a limit to keywords in the database. 
How many keywords do you have?

**Reply by lastzero** [reply]
I've checked it and there's no limit in the database, at least not just 250 characters or so. It's a text field. There may be limits on the way there, like when processing metadata to prevent denial of service attacks etc.

**Reply by bcdixit** [reply]
@graciousgrey , I have some photos from flickr that have 15 to 20 keywords (flickr tags). When I check in Photoprism, there seems to be some limit on the number of keywords it can accept.  Would converting flickr photo tag to photoprism labels be a better option?

**Reply by graciousgrey** [reply]
@bcdixit we are not sure why exactly it isn't working. 
You could try to add those directly in the database, but it might be that they are filtered for some reason (e.g. because keywords are on the stopwords list). 
Could you send us examples to samples@photoprism.app? Also some samples of the flickr JSON would be awesome, we plan to import from those in the future (https://github.com/photoprism/photoprism/issues/812)

**Comment by brownphotographic** [comment]
A useful link to how to get the free exiftool to scrape through the files from a flickr export and apply exif to the images accordingly:

https://exiftool.org/forum/index.php?topic=11072.0
